lib/interval/interval.ex

```diff
@@ -331,6 +331,83 @@ defmodule Timex.Interval do
     end
   end
 
+  @doc """
+  Removes one interval from another, which may reduce, split, or
+  eliminate the original interval. Returns a (possibly empty) list
+  of intervals representing the remaining time.
+
+  ## Examples
+
+  iex> #{__MODULE__}.difference(#{__MODULE__}.new(from: ~N[2018-01-01 02:00:00.000], until: ~N[2018-01-01 04:00:00.000]), #{__MODULE__}.new(from: ~N[2018-01-01 03:00:00.000], until: ~N[2018-01-01 05:00:00.000]))
+  {:ok, [%#{__MODULE__}{from: ~N[2018-01-01 02:00:00.000], left_open: true, right_open: false, step: [days: 1], until: ~N[2018-01-01 03:00:00.000]}]}
+
+  iex> #{__MODULE__}.difference(#{__MODULE__}.new(from: ~N[2018-01-01 01:00:00.000], until: ~N[2018-01-01 05:00:00.000]), #{__MODULE__}.new(from: ~N[2018-01-01 02:00:00.000], until: ~N[2018-01-01 03:00:00.000]))
+  {:ok, [%#{__MODULE__}{from: ~N[2018-01-01 01:00:00.000], left_open: true, right_open: false, step: [days: 1], until: ~N[2018-01-01 02:00:00.000]}, %#{__MODULE__}{from: ~N[2018-01-01 03:00:00.000], left_open: true, right_open: false, step: [days: 1], until: ~N[2018-01-01 05:00:00.000]}]}
+
+  iex> #{__MODULE__}.difference(#{__MODULE__}.new(from: ~N[2018-01-01 02:00:00.000], until: ~N[2018-01-01 04:00:00.000]), #{__MODULE__}.new(from: ~N[2018-01-01 01:00:00.000], until: ~N[2018-01-01 05:00:00.000]))
+  {:ok, []}
+  """
+  @spec difference(__MODULE__.t(), __MODULE__.t()) :: {:ok, [__MODULE__.t()]}
+  def difference(%__MODULE__{} = original, %__MODULE__{} = remove) do
+    cond do
+      contains?(remove, original) ->
+        # Cases like:
+        #
+        #   |- original -|
+        # |---  remove  ---|
+        #
+        # |- original -|
+        # |---  remove  ---|
+        #
+        #     |- original -|
+        # |---  remove  ---|
+        #
+        # |- original -|
+        # |-  remove  -|
```

Comment:
I found these little textual graphs helpful to understand the cases and ensure I'd covered them all. If you don't like keeping them as comments, we can delete them.